compiler/resolution/lowerIterators.cpp

```diff
@@ -1480,27 +1480,43 @@ expandBodyForIteratorInline(ForLoop*       forLoop,
                             TaskFnCopyMap& taskFnCopies,
                             bool&          addErrorArgToCall);
 
-static void markLoopProperties(ForLoop* forLoop, BlockStmt* ibody) {
-  bool isOrderIndependent = forLoop->isOrderIndependent();
-  bool hasVectorHazard = forLoop->hasVectorizationHazard();
+static void markLoopProperties(ForLoop* forLoop, BlockStmt* ibody,
+                               bool forVectorize) {
+  bool forIsOrderIndep = forLoop->isOrderIndependent();
+  bool forHasHazard = forLoop->hasVectorizationHazard();
```

Comment:
I'd move these two after the if below, and drop `forIsOrderIndep` from that block

Reply:
That would change the behavior. forVectorize is for the case that vectorize is explicitly requested (e.g. with `vectorizeOnly` and should ignore hazards.